File: app/src/main/java/com/veganbeauty/app/features/home/adapter/HomeShortcutAdapter.java
package com.veganbeauty.app.features.home.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.databinding.HomeItemShortcutBinding;

import java.util.ArrayList;
import java.util.List;

public class HomeShortcutAdapter extends RecyclerView.Adapter<HomeShortcutAdapter.ViewHolder> {

    private final List<HomeShortcutItem> items = new ArrayList<>();

    public void submitList(List<HomeShortcutItem> newItems) {
        items.clear();
        items.addAll(newItems);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        HomeItemShortcutBinding binding = HomeItemShortcutBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        );
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final HomeItemShortcutBinding binding;

        public ViewHolder(HomeItemShortcutBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(HomeShortcutItem item) {
            binding.tvShortcutName.setText(item.getTitle());
            binding.ivShortcutIcon.setImageResource(item.getIconResId());
            binding.getRoot().setOnClickListener(v -> item.getAction().run());
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/home/adapter/HomeShortcutAdapter.kt
package com.veganbeauty.app.features.home.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.veganbeauty.app.databinding.HomeItemShortcutBinding

data class HomeShortcutItem(
    val title: String,
    val iconResId: Int,
    val action: () -> Unit
)

class HomeShortcutAdapter : RecyclerView.Adapter<HomeShortcutAdapter.ViewHolder>() {

    private val items = mutableListOf<HomeShortcutItem>()

    fun submitList(newItems: List<HomeShortcutItem>) {
        items.clear()
        items.addAll(newItems)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = HomeItemShortcutBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int = items.size

    inner class ViewHolder(private val binding: HomeItemShortcutBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(item: HomeShortcutItem) {
            binding.tvShortcutName.text = item.title
            binding.ivShortcutIcon.setImageResource(item.iconResId)
            binding.root.setOnClickListener { item.action() }
        }
    }
}
